### PR TITLE
Add `test_pdbinput_input` unit test

### PR DESCRIPTION
**Pull request by amorehead**
* Adds the new `test_pdbinput_input` unit test that verifies functionality of training and sampling with a single PDB mmCIF file and visualizes the output (sampled) 3D coordinates in place of the input 3D coordinates within the original input mmCIF.
* Also treats modified polymer residues as N-atom ligands following the AF3 supplement.
* **NOTE**: The two `TODO` comments left in `input.py` denote two bugs (I believe) I've identified where padding at the end of a tensor (for ligands I presume) leads to either the wrong residue ID being set for padded tokens or leaves some tokens without any molecule type in `is_molecule_types`. These should be looked at and addressed in separate issues after merging this PR.

### CODE REVIEW

File: alphafold3_pytorch/inputs.py [hunk 1]
```diff
@@ -783,6 +819,7 @@ def alphafold3_input_to_molecule_input(
     # handle molecule ids
 
     molecule_ids = torch.cat(molecule_ids).long()
+    # TODO: do not pad this with zeros anymore, as it will mistakenly treat padded tokens as `ALA`
```
Comment (lucidrains) [inline alphafold3_pytorch/inputs.py:822]: oh i see, yea these are for the metal ions and eventual miscellaneous biomolecules

what id do you think it should have? is this that `gap_id` i saw earlier?

File: alphafold3_pytorch/inputs.py [hunk 2]
```diff
@@ -706,33 +744,32 @@ def alphafold3_input_to_molecule_input(
...
-    assert num_tokens > 0, f'you have an empty alphafold3 input'
+    assert num_tokens > 0, "you have an empty alphafold3 input"
 
     arange = torch.arange(num_tokens)[:, None]
 
-    molecule_types_lens_cumsum = tensor([0, *molecule_type_token_lens]).cumsum(dim = -1)
+    molecule_types_lens_cumsum = tensor([0, *molecule_type_token_lens]).cumsum(dim=-1)
     left, right = molecule_types_lens_cumsum[:-1], molecule_types_lens_cumsum[1:]
 
+    # TODO: fix bug that may leave molecules with no assigned type
```
Comment (lucidrains) [inline alphafold3_pytorch/inputs.py:759]: i think it is just the metal ions (which are last)

File: alphafold3_pytorch/alphafold3.py [hunk 3]
```diff
@@ -3377,9 +3378,10 @@ def forward(
         resolved_labels: Int['b n'] | None = None,
         return_loss_breakdown = False,
         return_loss: bool = None,
+        return_present_sampled_atoms: bool = False,
         num_rollout_steps: int = 20,
         rollout_show_tqdm_pbar: bool = False
-    ) -> Float['b m 3'] | Float[''] | Tuple[Float[''], LossBreakdown]:
+    ) -> Float['b m 3'] | Float['l 3'] | Float[''] | Tuple[Float[''], LossBreakdown]:
```
Comment (lucidrains) [inline alphafold3_pytorch/alphafold3.py:3384]: did you mean to do a `List[Float['l 3']]` here? (accounting for batched sampling of atom positions)

**Comment by lucidrains** [comment]
@amorehead looks great! some of the stuff is beyond me but at this point really trust what you are doing from the data end haha

File: alphafold3_pytorch/inputs.py (hunk 1 above)
Comment (amorehead) [reply alphafold3_pytorch/inputs.py:822]: If you don't know the type of a molecule, I'd say `gap_id=31` would be a safe bet to set these to.

File: alphafold3_pytorch/inputs.py (hunk 2 above)
Comment (amorehead) [reply alphafold3_pytorch/inputs.py:759]: Oh ok

File: alphafold3_pytorch/alphafold3.py (hunk 3 above)
Comment (amorehead) [reply alphafold3_pytorch/alphafold3.py:3384]: Nope, `l 3` means I sliced out the non-missing atom positions after sampling using `~missing_atom_mask`, which removes the batch dimension and thereby flattens the "present" atom positions into a 2D tensor
